common/lib/plugins/failover/reader_failover_handler.ts

```diff
@@ -207,7 +207,27 @@ export class ClusterAwareReaderFailoverHandler implements ReaderFailoverHandler
       tasks.push(secondTask.call());
     }
 
-    return await Promise.any(tasks);
+    return await Promise.any(tasks).catch((error: AggregateError) => {
+      let errors: string = "";
+      for (const e of error.errors) {
+        // Propagate errors that are not caused by network errors.
+        if (!this.pluginService.isNetworkError(error)) {
```

Comment:
What is the goal of this check here? Are we just checking to see if the aggregate error contains the AwsWrapperError we throw to indicate task aborted?
```
throw new AwsWrapperError("Selected task has been chosen. Abort client for host: " + this.newHost.host);
```

In this case would it be more clear if we just check for `error instanceof AwsWrapperError`?

Reply:
We could also be more specific and create a new `ReaderTaskAbortedError`

Reply:
In this check we want to ignore any network errors so we can continue trying connections to other instances (for example, `connect ECONNREFUSED`). If there are non-network errors, they will be propagated back to the user through the ReaderFailoverResult. This catch block should only be hit when both connection attempts in the batch failed, but the selected task error is only thrown when there has been a successful connection.